### DISCUSSION TITLE
Question about initial condition setting

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#24487] [Q&A Modules: Phase field]
Hi everyone,

I want to divide the whole mesh as a 3*3 grid. The initial condition of each small square should be phi=0+randomly generated value, 
how to set this initial condition?
(The following is the model I want)
![3fa4bfdfd5f2b192b35aed3d19d0364](https://github.com/idaholab/moose/assets/63679626/85a59d6e-9275-4917-8f72-b43a930a8b71)
I found that IsolatedBoundingBoxIC can divide the whole mesh to small square/rectangular and we can set different initial condition value to each small square/rectangular, however, it seems that this initial value can only be a certain value, it can not be set as randomly generated value in a range. How can I set a randomly generated value as the initial value in each small square/rectangular? 
(The following is IsolatedBoundingBoxIC)
![image](https://github.com/idaholab/moose/assets/63679626/3f27c0e4-f264-40f4-b60f-1688dcc228b5)

I also found that, using RandomIC can generate random value based on some distribution as the initial condition. How can I combine this with IsolatedBoundingBoxIC to set random generated value in each small box?
(The following is RandomIC)
![image](https://github.com/idaholab/moose/assets/63679626/cc5d965c-2aa7-4e44-8c3e-92953f5b18a4)

Thank you very much,
Best,

**Comment by GiudGiud** [comment]
Hello

we don’t have that one premade. You can code a randomBoxesIC that does what you want.

Guillaume

**Reply by abc-hy** [reply]
How to code a randomBoxesIC? Does this need any specific language?

**Reply by GiudGiud** [reply]
C++
MOOSE is open source and you can make any new object you want. This includes initial conditions
